library/include/modules/Screen.h

```diff
@@ -315,6 +322,8 @@ namespace DFHack
         namespace Hooks {
             GUI_HOOK_DECLARE(get_tile, Pen, (int x, int y, bool map, int32_t * df::graphic_viewportst::*texpos_field));
             GUI_HOOK_DECLARE(set_tile, bool, (const Pen &pen, int x, int y, bool map, int32_t * df::graphic_viewportst::*texpos_field));
+            // GUI_HOOK_DECLARE(get_tile_map_port, Pen, (int x, int y, int32_t * df::graphic_map_portst::*texpos_field));
+            GUI_HOOK_DECLARE(set_tile_map_port, bool, (const Pen &pen, int x, int y, int32_t * df::graphic_map_portst::*texpos_field));
```

Comment:
don't bother with this. This is a deprecated construct leftover from TWBT. You can just directly implement the new paint function without the GUI_HOOK redirection.